Native build can't find cabocha.h even though it is present

`gem install cabocha` doesn't appear to work because the native extensions can't be installed. Here's the output, using the specific version of Ruby's gem directly just to get rbenv out of the picture, since it has been blamed for this sort of thing in the past:

    $ ~/.rbenv/versions/2.1.6/bin/gem install cabocha
    Fetching: cabocha-0.69.gem (100%)
    Building native extensions.  This could take a while...
    ERROR:  Error installing cabocha:
    	ERROR: Failed to build gem native extension.
    
        /Users/trejkaz/.rbenv/versions/2.1.6/bin/ruby extconf.rb
    checking for main() in -lcabocha... no
    checking for main() in -lcrfpp... no
    checking for main() in -lmecab... yes
    checking for main() in -liconv... yes
    checking for main() in -l-L/usr/local/Cellar/mecab/0.996/lib... no
    checking for main() in -lmecab... yes
    checking for main() in -lstdc++... yes
    checking for cabocha.h... yes
    creating Makefile
    
    make "DESTDIR=" clean
    
    make "DESTDIR="
    compiling CaboCha_wrap.cpp
    CaboCha_wrap.cpp:1843:10: fatal error: 'cabocha.h' file not found
    #include "cabocha.h"
             ^
    1 error generated.
    make: *** [CaboCha_wrap.o] Error 1
    
    make failed, exit code 2
    
    Gem files will remain installed in /Users/trejkaz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/gems/cabocha-0.69 for inspection.
    Results logged to /Users/trejkaz/.rbenv/versions/2.1.6/lib/ruby/gems/2.1.0/extensions/x86_64-darwin-15/2.1.0-static/cabocha-0.69/gem_make.out

I went looking to see how it got the path and it seems to run `cabocha-config`, but when I run that tool myself, I get what looks like the right location:

    $ cabocha-config  --inc-dir
    /usr/local/Cellar/cabocha/0.69/include
    $ ls -la /usr/local/Cellar/cabocha/0.69/include
    total 32
    drwxr-xr-x   3 trejkaz  admin    102 24 Sep 23:35 .
    drwxr-xr-x  15 trejkaz  admin    510 18 Oct 15:50 ..
    -rw-r--r--   1 trejkaz  admin  12641 24 Sep 23:35 cabocha.h

Just to be doubly-sure, I made sure INCLUDE was set too:

    $ echo $INCLUDE
    /usr/local/include
    $ ls -l /usr/local/include/
    total 24
    lrwxr-xr-x   1 trejkaz  admin   40 18 Oct 15:50 cabocha.h -> ../Cellar/cabocha/0.69/include/cabocha.h

So I'm out of ideas. What's the trick?

Same result on 2.2.3. Reinstalling cabocha has no effect, but I had only just installed it last time, so I didn't really expect it to have an effect anyway.

```
$ /usr/local/var/rbenv/versions/2.2.3/bin/ruby  --version
ruby 2.2.3p173 (2015-08-18 revision 51636) [x86_64-darwin15]

$ /usr/local/var/rbenv/versions/2.2.3/bin/gem install cabocha
Fetching: cabocha-0.69.gem (100%)
Building native extensions.  This could take a while...
ERROR:  Error installing cabocha:
    ERROR: Failed to build gem native extension.

    /usr/local/var/rbenv/versions/2.2.3/bin/ruby -r ./siteconf20151023-50526-1npbzn0.rb extconf.rb
checking for main() in -lcabocha... no
checking for main() in -lcrfpp... no
checking for main() in -lmecab... yes
checking for main() in -liconv... yes
checking for main() in -l-L/usr/local/Cellar/mecab/0.996/lib... no
checking for main() in -lmecab... yes
checking for main() in -lstdc++... yes
checking for cabocha.h... yes
creating Makefile

make "DESTDIR=" clean

make "DESTDIR="
compiling CaboCha_wrap.cpp
CaboCha_wrap.cpp:1843:10: fatal error: 'cabocha.h' file not found
#include "cabocha.h"
         ^
1 error generated.
make: *** [CaboCha_wrap.o] Error 1

make failed, exit code 2

Gem files will remain installed in /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/cabocha-0.69 for inspection.
Results logged to /usr/local/var/rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/extensions/x86_64-darwin-15/2.2.0-static/cabocha-0.69/gem_make.out

$ gem install cabocha
(same, omitted)
```

--- a/cabocha.gemspec
+++ b/cabocha.gemspec
@@ -1,12 +1,8 @@
 # coding: utf-8
 
-lib = File.expand_path('../lib', __FILE__)
-$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
-require 'cabocha'
-
 Gem::Specification.new do |spec|
   spec.name          = 'cabocha'
-  spec.version       = CaboCha::VERSION
+  spec.version       = '0.69-1'
   spec.authors       = ['Yasuaki Uechi']
   spec.email         = ['uetchy@randompaper.co']
   spec.summary       = %q{cabocha-ruby is a gem that provides Ruby bindings for CaboCha.}

--- a/ext/cabocha/extconf.rb
+++ b/ext/cabocha/extconf.rb
@@ -7,6 +7,7 @@ enable_config('cabocha-config')
   have_library(lib)
 }
 
-$CFLAGS += ' ' + `#{cabocha_config} --cflags`.chomp
+$CPPFLAGS += ' ' + `#{cabocha_config} --cflags`.chomp
+$LDFLAGS += ' ' + `#{cabocha_config} --libs`.chomp
 
 have_header('cabocha.h') && create_makefile('cabocha/CaboCha')

